packages/edge-gateway/src/gateway.js

```diff
@@ -110,6 +110,13 @@ export async function gatewayGet (request, env, ctx) {
       const { status, reason } = JSON.parse(denyListResource)
       return new Response(reason || '', { status: status || 410 })
     }
+
+    if (winnerGwResponse.headers.get('content-type') === 'text/html') {
+      // we don't wait for google because it adds to the overall latency, but in the
+      // background we'll update KV so future requests coming in will properly be
+      // blocked as CIDs/malware are flagged.
+      fetchGoogleMalwareDetection(winnerUrl, cid, env)
+    }
```

Comment:
We need to add a safeguard here to prevent recursive requests (more info on https://hackmd.io/@vasco-santos/rJWZsz-Cc point 1 flow notes)